fili-core/src/main/java/com/yahoo/bard/webservice/druid/serializers/Util.java

```diff
@@ -24,24 +25,42 @@
      * @return  an Optional String of physical name
      */
     public static Optional<String> findPhysicalName(Dimension value, JsonGenerator gen) {
-        JsonStreamContext context = gen.getOutputContext();
         String apiName = value.getApiName();
         // Search for physical name
+        return withDruidQuery(gen, druidQuery -> druidQuery.getDataSource().getPhysicalTables().iterator().next().getPhysicalColumnName(apiName)
+        );
+    }
+
+    /**
+     * JSON tree walk to determine if there is a nested query below the current json node or not.
+     *
+     * @param gen  the Json Generator to retrieve the tree to walk on.
+     *
+     * @return  a Boolean where true indicates there are more nested query below this node, false otherwise
+     */
+    public static Boolean hasInnerQuery(JsonGenerator gen) {
+        return withDruidQuery(gen, druidQuery -> druidQuery.getInnerQuery() != null).orElse(false);
+    }
+
+    /**
+     * JSON tree walk to find the druid query context of the current context,
+     * returns the current context if current context is a druid query.
+     *
+     * @param gen  the Json Generator to retrieve the tree to walk on.
+     * @param success  a function that takes an DruidQuery as an argument and return the final desired returned result.
+     *
+     * @return an Optional of DruidQuery if found, Optional.empty otherwise
+     */
+    public static <T> Optional<T> withDruidQuery(JsonGenerator gen, Function<DruidQuery, T> success) {
+        JsonStreamContext context = gen.getOutputContext();
```

Comment:
withDruidQuery is a good name if you were extracting or building a druid query, but in this case you're mapping a value from a druid query.